ClearKeep/Scenes/InCall/CustomVideoView.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  CustomVideoView.swift
+//  ClearKeep
+//
+//  Created by Luong Minh Hiep on 25/05/2021.
+//
+
+import SwiftUI
+import WebRTC
+import ChatSecure
+
+struct CustomVideoView: View {
+	
+	@ObservedObject var videoViewConfig: CustomVideoViewConfig
+	let rtcVideoView: RTCMTLEAGLVideoView
+	
+	var body: some View {
+		GeometryReader { _ in
+			ZStack {
+				VideoView(rtcVideoView: rtcVideoView)
+					.rotation3DEffect(.degrees(180), axis: (x: 0, y: 1, z: 0))
+					.blur(radius: videoViewConfig.showOptionView ? 70 : 0)
+				
+				if videoViewConfig.showOptionView {
+					if videoViewConfig.microEnable {
+						Group {
+							VStack {
+								CallActionButtonView(onIcon: AppTheme.shared.imageSet.muteIcon,
+													 offIcon: AppTheme.shared.imageSet.muteIcon,
+													 isOn: true,
+													 title: "",
+													 styleButton: .video,
+													 widthButton: 48,
+													 widthInsideIcon: 22,
+													 action: {
+									videoViewConfig.microEnable.toggle()
+								})
+								
+								Text("Tap to mute")
+									.font(AppTheme.shared.fontSet.font(style: .body4))
+									.foregroundColor(AppTheme.shared.colorSet.grey4)
+									.padding(.bottom, 24)
+							}
+						}
+					} else {
+						Group {
+							VStack {
+								CallActionButtonView(onIcon: AppTheme.shared.imageSet.muteOffIcon,
+													 offIcon: AppTheme.shared.imageSet.muteOffIcon,
+													 isOn: true,
+													 title: "",
+													 styleButton: .endCall,
+													 widthButton: 48,
+													 widthInsideIcon: 22,
+													 action: {
+									videoViewConfig.microEnable.toggle()
+								})
+								
+								Text("Microphone off")
+									.font(AppTheme.shared.fontSet.font(style: .body4))
+									.foregroundColor(AppTheme.shared.colorSet.grey4)
+									.padding(.bottom, 24)
+							}
+						}
+					}
+				}
+				
+				if !videoViewConfig.showOptionView && !videoViewConfig.microEnable {
+					VStack {
+						Spacer()
+						HStack {
+							CallActionButtonView(onIcon: AppTheme.shared.imageSet.muteOffIcon,
+												 offIcon: AppTheme.shared.imageSet.muteOffIcon,
+												 isOn: true,
+												 title: "",
+												 styleButton: .endCall,
+												 widthButton: 48,
+												 widthInsideIcon: 22,
+												 action: {
+								videoViewConfig.microEnable.toggle()
+							})
+							.padding(.all, 2)
+							Spacer()
+						}
+					}
+					.opacity(0.64)
+				}
+				
+				VStack(alignment: .center) {
+					Spacer()
+					Text(videoViewConfig.userName)
+						.font(AppTheme.shared.fontSet.font(style: .display3))
+						.foregroundColor(AppTheme.shared.colorSet.offWhite)
+						.padding(.bottom, 24)
+				}
+			}
+			.clipShape(Rectangle())
+			.onTapGesture {
+				videoViewConfig.triggerOptionDisplayTimeout()
+			}
+		}
+	}
+	
+	
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/fe9f474e9dacdb67166a3b18715e917932515a0f/ClearKeep/Scenes/InCall/CustomVideoView.swift#L104"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>CustomVideoView.swift:104</code></td>
    </tr>
  </tbody>
</table>